### DISCUSSION TITLE
Anyone see this webpack(5) error during a nextjs build?   Cache group "shared" conflicts with existing chunk.

### DISCUSSION THREAD

**Discussion by SirensOfTitan** [vercel/next.js#18339] [Help]
Hitting into a frustrating error I'm unsure how to debug after a pretty innocuous refactor (moving some simple util functions into different files).  I don't want to create a bug report as I don't know how to reproduce reliably outside of my (quite large) project.

Full error:
```
SplitChunksPlugin
Cache group "shared" conflicts with existing chunk.
Both have the same name "c8f7fe3b0e41be846d5687592cf2018ff6e22687" and existing chunkf the selected modules.
Use a different name for the cache group or make sure that the existing chunk is a papendsOn).
HINT: You can omit "name" to automatically create a name.
BREAKING CHANGE: webpack < 5 used to allow to use an entrypoint as splitChunk. This id when the entrypoint is not a parent of the selected modules.
Remove this entrypoint and add modules to cache group's 'test' instead. If you need mated on startup, add them to the existing entrypoints (make them arrays). See migratinfo.
```

I've seen this on Next 9.5.x and next 10.x from webpack versions 5.1-5.3.  This occurs on a clean build (no webpack cache).

**Comment by timneutkens** [comment]
You most likely have customized the webpack configuration 🤔

**Comment by ramasilveyra** [comment]
@SirensOfTitan I'm seeing the same issue. Did you find a solution? (same context, webpack 5 and big project)

**Reply by ramasilveyra** [reply]
Solved by adding `config.optimization.splitChunks.cacheGroups.shared = false;` to next.config.js.

**Reply by ReneF1** [reply]
For me it says undefined - shared of undefined does not exists. Can you show me where you put it?

**Comment by gtempesta** [comment]
I had a very similar error message. I think it was caused by two modules in the same directory being dynamically imported in different parts of the code. One of the two imports didn't have a `webpackChunkName` magic comment, so the solution was to add the name in the import that didn't have it.

```js
if (!window.prod) {
  import('@/chunks/staff' /* webpackChunkName: "staff" */).then(() => {});
}
```
